Sources/CMakeLists.txt

```diff
@@ -12,12 +12,28 @@
 ##
 ##===----------------------------------------------------------------------===##
 
-add_subdirectory(_FoundationCShims)
-
-# Disable the macro build on Windows until we can correctly build it for the host architecture
-if(NOT CMAKE_SYSTEM_NAME STREQUAL Windows)
-	add_subdirectory(FoundationMacros)
+include(ExternalProject)
+if(CMAKE_HOST_WIN32)
+  set(_FoundationMacrosSwiftFlags "-use-ld=lld")
+endif()
+ExternalProject_Add(FoundationMacros
```

Comment:
Just to clarify with this approach, I know that the build script will need to invoke the macro build separately. Is the goal that it needs to invoke the macro build once for the current host (and pass that path to the swift-foundation build to use when building itself) and once for each target host (and that it will actually install instead of just building)? In `FoundationEssentials/CMakeLists.txt` we have a `FoundationEssentials` --> `FoundationMacros` dependency added on non-Windows, I think we'll need to adjust that accordingly if there's no longer a target built in this same project